### PR TITLE
Updated User Schema and Modified sign up route

### LINKED ISSUE #46: Finish up signup route for mentor/mentee
**Requirements for feature**
- Add fields from figma to mentor/mentee sign up routes - [PR comment](https://github.com/TritonSE/ALUM-Mobile-Application/pull/40/files#r1107640815)
- Add email validation for mentor - [PR comment](https://github.com/TritonSE/ALUM-Mobile-Application/pull/40/files#r1107630271)

**Any context for feature**
This PR is just to finish up the mentor/mentee sign-up route
 
**Resources / tips for development**
Checkout past PR - https://github.com/TritonSE/ALUM-Mobile-Application/pull/40

### PR DESCRIPTION
**Pull request by adhi0331**
### Tracking Info

Resolves #46 

Make sure your branch name conforms to: `<feature|staging|bugfix|...>/<username>/<3-4 word description separated by dashes>`. Otherwise, please rename your branch and create a new PR.

### Changes

What changes did you make?

I updated the user schema so that it stores all the relevant data that we need for our MVP. Also I changed the sign up routes so that all information gathered during sign up gets updated in the database. 

### Testing

How did you confirm your changes worked? 

I tested the routes on postman and was able to create both mentor's and mentee's. I also tested the following edge cases:

- missing parameters in request
- trying to signup using existing email
- checking to make sure password is at least 6 chars long (firebase requires this as a minimum)

Mentee Checking:

Screenshot 2023-02-18 at 6 22 06 PM

Screenshot 2023-02-18 at 6 24 21 PM

Screenshot 2023-02-18 at 6 24 38 PM

Mentor Checking: 

Screenshot 2023-02-18 at 6 22 33 PM

Screenshot 2023-02-18 at 6 22 59 PM

Screenshot 2023-02-18 at 6 23 18 PM

Screenshot 2023-02-18 at 6 24 06 PM

### Confirmation of Change 

Changes for the routes can be found in the routes/user.ts file. Changes for the schemas can be found in the models/mentee.ts and models/mentor.ts folder.

### CODE REVIEW

**Comment by adhi0331** [comment]
Is it possible to remove the lint check where we require that only objects of error type to be thrown. I created custom errors and I can't throw them because they are not of error type.

**Comment by AmanKAggarwal** [comment]
> Is it possible to remove the lint check where we require that only objects of error type to be thrown. I created custom errors and I can't throw them because they are not of error type. 

Can we just make sure all our custom errors extend Error? Does that solve the lint issue? If it does, I think it'll be a good practice to be compatible with the native `Error` type. If you have strong inclination towards not doing that, lmk and we can look into disabling it.

**Comment by adhi0331** [comment]
> 

I have it set up that way right now, but that isn't fixing the issue.

**Comment by AmanKAggarwal** [comment]
```
✖ 6 problems (0 errors, 6 warnings)

Checking formatting...
[warn] src/errors/errors.ts
[warn] src/errors/validationError.ts
[warn] src/middleware/validation.ts
[warn] src/models/mentee.ts
[warn] src/models/mentor.ts
[warn] src/routes/user.ts
[warn] src/services/auth.ts
[warn] Code style issues found in 7 files. Forgot to run Prettier?
Error: Process completed with exit code 1.
```

From the error message, you should be able to resolve the linting errors by just running `npm run lint-fix` because it is failing due to code styling issues.

**Comment by AmanKAggarwal** [comment]
Great work with the screenshots! helps a lot with review

**Review by YashRavipati1 — APPROVED** [review]
Everything looks good to me

File: backend/src/middleware/validation.ts [hunk 1]
```diff
@@ -11,33 +11,41 @@
...
 /**
  * mentee cake type to be used to validate mentee
  * NOTE: proper request will follow this format
  */
 const MENTEE = bake({
-  type: string,
   name: string,
   email: string,
   password: string,
+  grade: string,
```
[the remark this answers is not in the document]
Comment (adhi0331) [reply backend/src/middleware/validation.ts:26]: Yeah I wasn't to sure if grade should be a string or number. I can change it to number if other's feel that will make more sense. @AmanKAggarwal Any thoughts?

[the remark this answers is not in the document]
Comment (AmanKAggarwal) [reply backend/src/middleware/validation.ts:26]: Yeah let's make grade a number and get this PR in!!

**Review by christen03 — APPROVED** [review]
